components/layout/table_wrapper.rs

```diff
@@ -778,7 +778,13 @@ fn initial_computed_inline_size(block: &mut BlockFlow,
                                                        containing_block_inline_size);
     match inline_size_from_style {
         MaybeAuto::Auto => {
-            MaybeAuto::Specified(min(containing_block_inline_size, preferred_width_of_all_columns))
+            if preferred_width_of_all_columns + table_border_padding <= containing_block_inline_size {
```

Comment:
I thought that the idea of the preferred width is that nothing should be greater than it.

That being said, table layout is hard, and I'm probably not the best person to review this.